src/code/z_game_dlftbls.c

```diff
@@ -1,18 +1,25 @@
 #include "global.h"
 
-#define GAMESTATE_OVERLAY(name, init, destroy, size)                                                     \
-    {                                                                                                    \
-        NULL, (uintptr_t)_ovl_##name##SegmentRomStart, (uintptr_t)_ovl_##name##SegmentRomEnd,            \
-            _ovl_##name##SegmentStart, _ovl_##name##SegmentEnd, NULL, init, destroy, NULL, NULL, 0, size \
-    }
-#define GAMESTATE_OVERLAY_INTERNAL(init, destroy, size) \
-    { NULL, 0, 0, NULL, NULL, NULL, init, destroy, NULL, NULL, 0, size }
+#define DEFINE_GAMESTATE_INTERNAL(typeName, enumName) \
+    { NULL, 0, 0, NULL, NULL, NULL, typeName##_Init, typeName##_Destroy, NULL, NULL, 0, sizeof(typeName##State) },
+
+#define DEFINE_GAMESTATE(typeName, enumName, name) \
+    { NULL,                                        \
+      (uintptr_t)_ovl_##name##SegmentRomStart,     \
+      (uintptr_t)_ovl_##name##SegmentRomEnd,       \
+      _ovl_##name##SegmentStart,                   \
+      _ovl_##name##SegmentEnd,                     \
+      NULL,                                        \
+      typeName##_Init,                             \
+      typeName##_Destroy,                          \
+      NULL,                                        \
+      NULL,                                        \
+      0,                                           \
+      sizeof(typeName##State) },
 
 GameStateOverlay gGameStateOverlayTable[] = {
-    GAMESTATE_OVERLAY_INTERNAL(Setup_Init, Setup_Destroy, sizeof(SetupState)),
-    GAMESTATE_OVERLAY(select, MapSelect_Init, MapSelect_Destroy, sizeof(MapSelectState)),
-    GAMESTATE_OVERLAY(title, ConsoleLogo_Init, ConsoleLogo_Destroy, sizeof(ConsoleLogoState)),
-    GAMESTATE_OVERLAY_INTERNAL(Play_Init, Play_Destroy, sizeof(PlayState)),
-    GAMESTATE_OVERLAY(opening, TitleSetup_Init, TitleSetup_Destroy, sizeof(TitleSetupState)),
-    GAMESTATE_OVERLAY(file_choose, FileSelect_Init, FileSelect_Destroy, sizeof(FileSelectState)),
+#include "tables/gamestate_table.h"
 };
+
+#undef DEFINE_GAMESTATE
+#undef DEFINE_GAMESTATE_INTERNAL
```

Comment:
You may have left it out on purpose, but I believe we should be able to add another set of defines/include at the top of this file to declare the segment symbols required by the table? (similarly to how we have it for `z_actor_dlftbls.c` and such)
This would further automate the segment declaration part, and assuming the symbols are only used in this table we could remove the DECLARE_OVERLAY_SEGMENT lines currently in `segment_symbols.h`